docs/interchange.rst

```diff
@@ -614,6 +666,138 @@ Population requirements
 
 There are no requirements on a population table.
 
+.. _sec_provenance_requirements:
+
+Provenance requirements
+-----------------------
+
+There are no requirements on a provenance table.
+
+
+.. _sec_table_transformations:
+
+Table transformation methods
+============================
+
+The following methods operate *in place* on a :class:`TableCollection`,
+transforming them while preserving information.
+In some applications, tables may most naturally be produced in a way that is
+logically consistent, but not meeting all the requirements for validity that
+are established for algorithmic and efficiency reasons.
+These methods (while having other uses), can be used to make such a set of
+tables valid, and thus ready to be loaded into a tree sequence.
+
+This section is best skipped unless you are writing a program that records
+tables directly.
```

Comment:
We could alternatively put this section somewhere else, labeled "forwards time recording methods".  But, it's nice to have it here, because it does explain why we depend on some properties of `sort` and `simplify`.

Reply:
One point of this section is to explain what requirements on tables can be relaxed for the following methods.